File: API/Core/Models/Card.cs
using System.Text.Json.Serialization;

namespace Core.Models;

public class Card
{
    public int Id { get; set; }
    public string FrontText { get; set; }
    public string BackText { get; set; }
    public int StatusId { get; set; }
    public CardStatus Status { get; set; }
    public int DeckId { get; set; }
    [JsonIgnore]
    public Deck Deck { get; set; }
}

File: mobile/lib/models/card.dart
class Card {
  final int id;
  final String frontText;
  final String backText;
  final String status;

  const Card({
    required this.id,
    required this.frontText,
    required this.backText,
    required this.status,
  });

  factory Card.fromJson(Map<String, dynamic> json) {
    return Card(
        id: json['id'],
        frontText: json['frontText'],
        backText: json['backText'],
        status: json['status']);
  }
}
